From a485c928ee9872c8ce7ddbb8e9ba6f33f9ef23cc Mon Sep 17 00:00:00 2001
From: "Alex Xu (Hello71)" <alex_y_xu@yahoo.ca>
Date: Tue, 28 Jan 2025 17:17:01 -0500
Subject: [PATCH] meson: generate keyboard-keys-list from local input.h

otherwise it will use the system input.h which will fail to build if
newer than the bundled version

Fixes: 0a73c8e7b8 ("linux: import input.h and friends")
(cherry picked from commit bc996fd1ba7bce7be4cbae0d0d5ba7c35e8c3f50)
---
 src/udev/generate-keyboard-keys-list.sh | 2 +-
 src/udev/meson.build                    | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/udev/generate-keyboard-keys-list.sh b/src/udev/generate-keyboard-keys-list.sh
index ead3113342..4cac507322 100755
--- a/src/udev/generate-keyboard-keys-list.sh
+++ b/src/udev/generate-keyboard-keys-list.sh
@@ -3,7 +3,7 @@
 set -eu
 set -o pipefail
 
-${1:?} -dM -include linux/input.h - </dev/null | awk '
+${1:?} -dM -include "${2:?}" - </dev/null | awk '
     /\<(KEY_(MAX|MIN_INTERESTING))|(BTN_(MISC|MOUSE|JOYSTICK|GAMEPAD|DIGI|WHEEL|TRIGGER_HAPPY))\>/  { next }
     /^#define[ \t]+(KEY|BTN)_[^ ]+[ \t]+[0-9BK]/                                                    { print $2 }
 '
diff --git a/src/udev/meson.build b/src/udev/meson.build
index 3535551e74..921dfac39c 100644
--- a/src/udev/meson.build
+++ b/src/udev/meson.build
@@ -57,8 +57,9 @@ endif
 generate_keyboard_keys_list = find_program('generate-keyboard-keys-list.sh')
 keyboard_keys_list_txt = custom_target(
         'keyboard-keys-list.txt',
+        input : files('../basic/linux/input.h'),
         output : 'keyboard-keys-list.txt',
-        command : [generate_keyboard_keys_list, cpp],
+        command : [generate_keyboard_keys_list, cpp, '@INPUT@'],
         capture : true)
 
 generate_keyboard_keys_gperf = find_program('generate-keyboard-keys-gperf.sh')
